mmdet/models/dense_heads/yolo_head.py

```diff
@@ -197,30 +197,28 @@ def get_bboxes(self,
                 (n,) tensor where each item is the predicted class label of the
                 corresponding box.
         """
-        result_list = []
         num_levels = len(pred_maps)
-        for img_id in range(len(img_metas)):
-            pred_maps_list = [
-                pred_maps[i][img_id].detach() for i in range(num_levels)
-            ]
-            scale_factor = img_metas[img_id]['scale_factor']
-            proposals = self._get_bboxes_single(pred_maps_list, scale_factor,
-                                                cfg, rescale, with_nms)
-            result_list.append(proposals)
+        pred_maps_list = [pred_maps[i].detach() for i in range(num_levels)]
+        scale_factors = [
+            img_metas[i]['scale_factor']
+            for i in range(pred_maps_list[0].shape[0])
+        ]
+        result_list = self._get_bboxes(pred_maps_list, scale_factors, cfg,
+                                       rescale, with_nms)
         return result_list
 
-    def _get_bboxes_single(self,
-                           pred_maps_list,
-                           scale_factor,
-                           cfg,
-                           rescale=False,
-                           with_nms=True):
+    def _get_bboxes(self,
```

Comment:
1 comments the length of the list
2 (num_points * num_classes, H, W) in docstr maybe wrong?